api/src/test/java/org/openmrs/notification/MessageServiceTest.java

```diff
@@ -7,110 +7,294 @@
+	Context context = new Context();
+	context.setServiceContext(mockServiceContext);
 	
-	private static final String NO_SMTP_SERVER_ERROR = "Could not connect to SMTP host:";
-	
-	MessageService ms = null;
-	
-	/**
-	 * Run this before each unit test in this class. The "@Before" method in
-	 * {@link BaseContextSensitiveTest} is run right before this method.
-	 * 
-	 * @throws Exception
-	 */
-	@BeforeEach
-	public void runBeforeEachTest() {
-		executeDataSet("org/openmrs/notification/include/MessageServiceTest-initial.xml");
-		
-		ms = Context.getMessageService();
-	}
 	
-	/**
-	 * @throws MessageException
-	 * @see MessageService#createMessage(String,String,String,String)
-	 */
+	mockedUserList = new ArrayList<>();
+    User user1 = new User();
+    user1.setEmail("user@example.com");
+    mockedUserList.add(user1);
+
+      lenient().when(userService.getUsers(any(), anyList(), anyBoolean()))
+    .thenReturn(mockedUserList);
+
+
+    executeDataSet("org/openmrs/notification/include/MessageServiceTest-initial.xml");
+}
+
+    /**
+     * Verifies that a message is correctly created with all its fields populated.
+     *
+     * @throws MessageException if message creation fails
+     */
 	@Test
-	public void createMessage_shouldCreateMessage() throws MessageException {
-		String recipients = "foo@bar.com,marco@polo.com";
+	public void createMessage_shouldCreateMessageWithAllFields() throws MessageException {
+		// Arrange
+		String recipients = "foo@bar.com, marco@polo.com";
 		String sender = "me@mydomain.com";
-		String subject = "foo";
-		String message = "content";
-		String attachment = "inga";
+		String subject = "Test Subject";
+		String content = "This is a test message.";
+		String attachment = "testfile";
 		String attachmentContentType = "text/plain";
-		String attachmentFileName = "inga.txt";
-		
-		Message msg1 = ms.createMessage(recipients, sender, subject, message);
-		Message msg2 = ms.createMessage(subject, message);
-		Message msg3 = ms.createMessage(sender, subject, message);
-		Message msg4 = ms.createMessage(recipients, sender, subject, message, attachment, attachmentContentType,
-		    attachmentFileName);
-		
-		assertEquals(recipients, msg1.getRecipients());
-		assertEquals(recipients, msg4.getRecipients());
-		
-		assertEquals(sender, msg1.getSender());
-		assertEquals(sender, msg3.getSender());
-		assertEquals(sender, msg4.getSender());
-		
-		assertEquals(subject, msg1.getSubject());
-		assertEquals(subject, msg2.getSubject());
-		assertEquals(subject, msg3.getSubject());
-		assertEquals(subject, msg4.getSubject());
-		
-		assertEquals(message, msg1.getContent());
-		assertEquals(message, msg2.getContent());
-		assertEquals(message, msg3.getContent());
-		assertEquals(message, msg4.getContent());
-		
-		assertEquals(attachment, msg4.getAttachment());
-		assertEquals(attachmentContentType, msg4.getAttachmentContentType());
-		assertEquals(attachmentFileName, msg4.getAttachmentFileName());
```

Comment:
I'm fine with re-writing the tests into a single-code-path-per test. That makes sense. However, you've now actually removed coverage for everything except one function. We should also have a test for the other variations of `createMessage()`, e.g., `createMessage_shouldCreateMessageWithJustSubjectAndMessage()`, etc.